jfix-armeria-aggregating-profiler/src/main/kotlin/ru/fix/armeria/aggregating/profiler/ProfiledError.kt

```diff
@@ -63,12 +62,16 @@ internal sealed class ProfiledError(
     }
 }
 
-internal fun RequestLog.detectProfiledErrorIfAny(): ProfiledError? {
+/**
+ * @param isResponseStatusValid  It is guaranteed that input [HttpStatus] is not [HttpStatus.UNKNOWN]
+ * so that if this checking function is called then http response is received
+ */
+internal fun RequestLog.detectProfiledErrorIfAny(isResponseStatusValid: (HttpStatus) -> Boolean): ProfiledError? {
     val requestCause = requestCause()
     val responseCause = responseCause()
     val status = responseHeaders().status()
     return when {
-        status.codeClass() != HttpStatusClass.SUCCESS && status != HttpStatus.UNKNOWN -> ProfiledError.InvalidStatus
+        status != HttpStatus.UNKNOWN && !isResponseStatusValid(status) -> ProfiledError.InvalidStatus
```

Comment:
Maybe we should also put check for unknown status into isResponseStatusValid method instead of separate comparsion here?

Reply:
https://github.com/ru-fix/jfix-armeria/pull/23#discussion_r626367771